examples/autoqasm/3_Iterative_phase_estimation.ipynb

```diff
@@ -82,13 +82,12 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "@aq.main(num_qubits=2)\n",
-    "def ipe(n_iterations: int):\n",
-    "    \"\"\"Iterative phase estimation algorithm.\n",
+    "# TODO: consider using n_iterations as an input (blocked by #818)\n",
```

Comment:
Yes, this is us todoing; I don't intend for the todo to become a part of the notebook. As written, with `n_iterations` outside of the function scope matches the functionality of the example before this change.

However, if a customer wanted to make a version of the program that worked for multiple values of `n_iterations`, the most accessible and imo recommended approach would be to use an input variable, rather than build a meta-programming wrapper around the program. I think this could be a good place to showcase that functionality

Reply:
Since this would break today, I'm open to dismissing this option for this PR, though maybe we should consider the update with https://github.com/amazon-braket/amazon-braket-sdk-python/pull/820, which fixes #818

I'll remove the todo here